diff --git a/rslib/src/storage/note/mod.rs b/rslib/src/storage/note/mod.rs
index 554753717..9fc40fa5c 100644
--- a/rslib/src/storage/note/mod.rs
+++ b/rslib/src/storage/note/mod.rs
@@ -70,10 +70,14 @@ impl super::SqliteStorage {
     }
 
     pub(crate) fn add_note(&self, note: &mut Note) -> Result<()> {
-        assert_eq!(note.id.0, 0);
+        // <<< assert_eq!(note.id.0, 0);
+        let id = match note.id.0 {
+            0 => TimestampMillis::now().into(),
+            id => id,
+        };
         let mut stmt = self.db.prepare_cached(include_str!("add.sql"))?;
         stmt.execute(params![
-            TimestampMillis::now(),
+            id,
             note.guid,
             note.notetype_id,
             note.mtime,
diff --git a/rslib/src/storage/notetype/mod.rs b/rslib/src/storage/notetype/mod.rs
index 692b68887..4f34b817c 100644
--- a/rslib/src/storage/notetype/mod.rs
+++ b/rslib/src/storage/notetype/mod.rs
@@ -245,13 +245,17 @@ impl SqliteStorage {
     }
 
     pub(crate) fn add_notetype(&self, nt: &mut Notetype) -> Result<()> {
-        assert_eq!(nt.id.0, 0);
+        // <<< assert_eq!(nt.id.0, 0);
+        let id = match nt.id.0 {
+            0 => TimestampMillis::now().into(),
+            id => id,
+        };
 
         let mut stmt = self.db.prepare_cached(include_str!("add_notetype.sql"))?;
         let mut config_bytes = vec![];
         nt.config.encode(&mut config_bytes)?;
         stmt.execute(params![
-            TimestampMillis::now(),
+            id,
             nt.name,
             nt.mtime_secs,
             nt.usn,
